pyriemann/utils/base.py

```diff
@@ -1,25 +1,36 @@
+"""Base functions for SPD matrices."""
+
 import numpy as np
 from scipy.linalg import eigh
 
 from numpy.core.numerictypes import typecodes
 
 
 def _matrix_operator(C, operator):
-    """Matrix equivalent of an operator."""
+    """Matrix function."""
     if C.dtype.char in typecodes['AllFloat'] and not np.isfinite(C).all():
         raise ValueError(
-            "Covariance matrices must be positive definite. Add "
+            "Matrices must be positive definite. Add "
             "regularization to avoid this error.")
     eigvals, eigvects = eigh(C, check_finite=False)
```

Comment:
you can actually call eigh on a 3D array to avoid the for loops and going through lists and arrays. See:

```
In [12]: size = 30

In [13]: C = np.random.random((3, size, size))

In [14]: evals, evecs = np.linalg.eig(C)

In [15]: evals.shape
Out[15]: (3, 30)

In [16]: evecs.shape
Out[16]: (3, 30, 30)

In [17]: np.testing.assert_allclose(evecs[0], np.linalg.eig(C[0])[1])

```

Reply:
You are right.
[numpy-based eigh](https://numpy.org/doc/stable/reference/generated/numpy.linalg.eigh.html) allows to process `(…, M, M) array`, but without `check_finite` option;
[scipy-based eigh](https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.eigh.html) treats only `(M, M) array`, but with `check_finite` option.
I don't know what is the best option.